docs/src/components/productX/XTreeViewDemo.tsx

```diff
@@ -1,261 +1,276 @@
+  UseTreeItem2Parameters,
+} from '@mui/x-tree-view/useTreeItem2';
+import {
+  TreeItem2Content,
+  TreeItem2IconContainer,
+  TreeItem2Label,
+  TreeItem2Root,
+} from '@mui/x-tree-view/TreeItem2';
+import { TreeItem2Icon } from '@mui/x-tree-view/TreeItem2Icon';
+import { TreeItem2Provider } from '@mui/x-tree-view/TreeItem2Provider';
+import { TreeViewBaseItem } from '@mui/x-tree-view/models';
 
-const CustomContent = React.forwardRef(function CustomContent(
-  props: TreeItemContentProps & { lastNestedChild?: boolean },
-  ref,
-) {
-  const {
-    lastNestedChild,
-    classes,
-    className,
-    label,
-    nodeId,
-    icon: iconProp,
-    expansionIcon,
-    displayIcon,
-  } = props;
+type FileType = 'image' | 'pdf' | 'video' | 'folder';
 
-  const {
-    disabled,
-    expanded,
-    selected,
-    focused,
-    handleExpansion,
-    handleSelection,
-    preventSelection,
-  } = useTreeItem(nodeId);
+type ExtendedTreeItemProps = {
+  fileType?: FileType;
+  id: string;
+  label: string;
+};
 
-  const icon = iconProp || expansionIcon || displayIcon;
+const ITEMS: TreeViewBaseItem<ExtendedTreeItemProps>[] = [
+  {
+    id: '1',
+    label: 'Drive',
+    children: [
+      {
+        id: '1.1',
+        label: 'Backup',
+        children: [
+          { id: '1.1.1', label: 'Jan 2023.pdf', fileType: 'pdf' },
+          { id: '1.1.2', label: 'Feb 2023.pdf', fileType: 'pdf' },
+          { id: '1.1.3', label: 'Mar 2023.pdf', fileType: 'pdf' },
+        ],
+      },
+      {
+        id: '1.2',
+        label: 'Photos',
+        children: [
+          { id: '1.2.1', label: 'family.jpeg', fileType: 'image' },
+          { id: '1.2.2', label: 'my_dog.png', fileType: 'image' },
+        ],
+      },
+    ],
+  },
+  {
+    id: '2',
+    label: 'Favorites',
+    children: [
+      {
+        id: '2.1',
+        label: 'MUI_retreat_photo.jpg',
+        fileType: 'image',
+      },
+      {
+        id: '2.2',
+        label: 'v7_secrets.mkv',
+        fileType: 'video',
+      },
+      {
+        id: '2.3',
+        label: 'Other pictures',
+        children: [{ id: '2.3.1', label: 'my_avatar.jpeg', fileType: 'image' }],
+      },
+    ],
+  },
+];
+
+declare module 'react' {
+  interface CSSProperties {
+    '--tree-view-color'?: string;
+    '--tree-view-bg-color'?: string;
+  }
+}
+
+const StyledTreeItemRoot = styled(TreeItem2Root)(({ theme }) => ({
+  color: theme.palette.mode === 'light' ? theme.palette.grey[800] : theme.palette.grey[400],
+  position: 'relative',
+  [`& .${treeItemClasses.groupTransition}`]: {
+    marginLeft: theme.spacing(3.5),
+  },
+})) as unknown as typeof TreeItem2Root;
 
-  const handleMouseDown = (event: React.MouseEvent<HTMLDivElement>) => {
-    preventSelection(event);
-  };
+const CustomTreeItemContent = styled(TreeItem2Content)(({ theme }) => ({
+  borderRadius: theme.spacing(0.7),
+  marginBottom: theme.spacing(0.5),
+  marginTop: theme.spacing(0.5),
+  padding: theme.spacing(0.5),
```

Comment:
```suggestion
  borderRadius: theme.spacing(0.5),
  marginBottom: theme.spacing(0.2),
  marginTop: theme.spacing(0.2),
```

I'd fine-tune these values a bit and remove the unnecessary padding!